src/GraphQL/TypeApi.hs

```diff
@@ -245,40 +250,47 @@ instance forall typeName interfaces fields m.
   buildResolver mHandler selectionSet = do
     -- First we run the actual handler function itself in IO.
     handler <- mHandler
-    -- We're evaluating an Object so we're collecting (name, Value)
-    -- pairs from runFields and build a GValue.Map with them.
-    r <- forM selectionSet $ \selection -> runFields @m @fields handler selection
-    pure $ GValue.toValue (GValue.mapFromList r)
+    -- We're evaluating an Object so we're collecting ObjectFields from
```

Comment:
The code that merges objects still needs to do _something_ when it gets duplicate fields. Am happy to add a TODO but don't know what that TODO would be.